capstone_project/src/main/java/com/google/sps/servlets/MapServlet.java

```diff
@@ -75,20 +73,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    addToDatastore(request);
+    School requestedSchool = new Gson().fromJson(request.getReader(), School.class);
+    addToDatastore(requestedSchool);
 
     // Respond with the result.
-    response.sendRedirect("pages/maps.html");
+    response.sendRedirect("/pages/maps.html");
   }
 
-  public void addToDatastore(HttpServletRequest request) throws IOException {
+  public void addToDatastore(School requestedSchool) throws IOException {
     // Initialize datastore object.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    // Get the input from the school entry form.
-    String schoolName = getParameter(request, "name-input");
-    double schoolLatitude = Double.parseDouble(getParameter(request, "latitude-input"));
-    double schoolLongitude = Double.parseDouble(getParameter(request, "longitude-input"));
+    // Parse the requestedSchool.
```

Comment:
I am so sorry, I thought this was a doGet method. the name submisssion makes more sense now. Can you change it back.

Reply:
no worries, I just changed it back! @Leoism